### ISSUE TITLE
Add BOKU drought vulnerability & crop-growth potential to GTIF AT

### ISSUE THREAD

**Issue by bschumac** [eurodatacube/eodash#2471]
EODC was tasked to suggest this new datasets as new layers in GTIF:

[Drought vulnerability](https://radiantearth.github.io/stac-browser/#/external/stac.eodc.eu/api/v1/collections/drought-vulnerability-2003-2018)

[Crop growth potential](https://radiantearth.github.io/stac-browser/#/external/stac.eodc.eu/api/v1/collections/intra-field-crop-growth-potential-2017-2020)

Find the descriptions of the layers in the STAC descriptions. Please note that BOKU requested to add explicitly a contact field. For Intra-field crop growth potential it is Contact: [francesco.vuolo@boku.ac.at](mailto:francesco.vuolo@boku.ac.at) or [clement.atzberger@boku.ac.at](mailto:clement.atzberger@boku.ac.at) and for Drought vulnerability it is Contact: [clement.atzberger@boku.ac.at](mailto:clement.atzberger@boku.ac.at)

Please also include the BOKU logo at the end of the description page.

**Comment by patrick-griffiths** [comment]
To be integrated as: 
- EO Adaptation Services > Agriculture Analysis > Drought Vulnerability
- EO Adaptation Services > Agriculture Analysis > Crop Growth Potential

**Comment by santilland** [comment]
Hello @bschumac it seems that the referenced datasets do not allow cross origin requests, so it would mean we would need to copy the data to a new location. Would it be possible to set the header to allow cross origin requests?

**Comment by senmao** [comment]
Hi @santilland  we have enabled the CORS for GTIF website as shown below. please try it again and let us know if it works for you.

access-control-allow-methods: GET
access-control-allow-origin: https://gtif.esa.int

**Comment by santilland** [comment]
Hello @senmao, thanks for adding CORS headers for GTIF.
CC @bschumac :
I tried to integrate the data now into the client, and was not getting it to work, after further inspection it seems that the files are not valid Cloud optimized Geotiffs (COG) as they have no overviews (have tiles). Testing with rio-cogeo validate:
```
The file is greater than 512xH or 512xW, but is not tiled
eodc/201210421_yield_potential_2017_2020_v8.tif is NOT a valid cloud optimized GeoTIFF
```

**Comment by santilland** [comment]
@bschumac  @senmao 
Trying to load the drought-vulnerability-2003-2018 data (which is small enough to not need tiles/overviews, but the requests fail.
When doing the request i get a 416 response error with the message "detail: "range exceeded file end".

If i download the file and host it locally everything works as expected. 

Here is a screenshot of the response headers from your server:
![image](https://github.com/eurodatacube/eodash/assets/4036819/69391c76-9a0f-47c5-a212-7dd3d1946a99)
here are response headers from serving the same file from a dev server locally:
![image](https://github.com/eurodatacube/eodash/assets/4036819/2f37b851-9cb5-4da9-a0c3-0294a10a2872)

If i interpret things right it seems that the content length is reported incorrectly on your server - content-length 34 in comparison with content length 42407 from the dev server. The file is 42K big.
 
Here is an example curl request:
curl -r 0-65536 https://data.eodc.eu/collections/GTIF/MCD13A2.Y20032018.006.globalV1.1_km_10_days_NDVI.O4.VCI.lethr35_1327_ge09_AT.tif --output test
Looking into the output you can see it shows the range request exceeds file end.

Maybe allowing range requests needs to be enabled for the server? I can't quite understand how the endpoint is configured.

**Comment by senmao** [comment]
The range request is already supported by our server. As sending the HEAD request to the data, you can get the content-length correctly 42407 as shown below. 
![image](https://github.com/eurodatacube/eodash/assets/6135112/87481d82-700a-439a-82f7-3d094065e72d)

So the requesting rang should not exceed the content length. I don't know why you got the content-length 34, it would be helpful if you can show the header of request, not only of the response.

**Comment by senmao** [comment]
> Hello @senmao, thanks for adding CORS headers for GTIF. CC @bschumac : I tried to integrate the data now into the client, and was not getting it to work, after further inspection it seems that the files are not valid Cloud optimized Geotiffs (COG) as they have no overviews (have tiles). Testing with rio-cogeo validate:
> 
> ```
> The file is greater than 512xH or 512xW, but is not tiled
> eodc/201210421_yield_potential_2017_2020_v8.tif is NOT a valid cloud optimized GeoTIFF
> ```

For the file format, I need to check my colleague who prepared this this data.

**Comment by senmao** [comment]
@santilland yes, that can one possible reason. however, in order to find out the root cause it would be good to have clear understanding how the frontside communicating with our backend. So if you can monitor all the network traffic to our server and note down the request header sent from frontend and the response header which returned from the server, we can know what causes the issue.

In general, I normally first send HEAD request to server to get header information. Then I can know if it supports range request and the content-length.

**Comment by santilland** [comment]
We are using OpenLayers Geotiff source to load and visualize the data, in the docs i see that the default blocksize is the one that goes over the current file size:
https://openlayers.org/en/latest/apidoc/module-ol_source_GeoTIFF.html#~GeoTIFFSourceOptions
(defaults to 65536) 
I have been experimenting with changing the block size to be smaller, but it somehow decides to only fetch one part and then stops... if i try to use the same block size as content size i get a strange NS_ERROR_NET_PARTIAL_TRANSFER 
Sometimes it sends an OPTIONS request before but id don't know exactly when, with that one i have issues with cors testing locally "CORS Preflight Did Not Succeed".
It does not first seem to send a HEAD request in any case, here is how it tries to load the data if i use the expected file size:
![image](https://github.com/eurodatacube/eodash/assets/4036819/359b1358-d338-43d9-b4f0-ab775629c021)
Not sure how to tackle this.

**Comment by senmao** [comment]
@santilland ok, to make things easy, I reset then range end to the file size on the server side when the range end exceeds the file size. so now it works if you send a range request of which the range end is larger than file size.

![image](https://github.com/eurodatacube/eodash/assets/6135112/5375f1fd-8357-40a5-9049-91322a65b953)

**Comment by santilland** [comment]
Great! Thank you @senmao that solves the issue and allows to load the data for the drought vulnerability

![image](https://github.com/eurodatacube/eodash/assets/4036819/52f8b2ef-5d3e-423f-87f6-4f66a80a1027)

**Comment by bschumac** [comment]
[Product_data_descriptors_BOKU_crop_growth_potential.docx](https://github.com/eurodatacube/eodash/files/14727554/Product_data_descriptors_BOKU_crop_growth_potential.docx)
[Product_data_descriptors_BOKU_drought-vulnerability.docx](https://github.com/eurodatacube/eodash/files/14727555/Product_data_descriptors_BOKU_drought-vulnerability.docx)


Here are the two data descriptors for the two requested datasets. Others will follow on 22/03/24.

**Comment by Patrick1G** [comment]
@santilland whats the status on these products? Can we integrate then in the live version??

**Comment by santilland** [comment]
@Patrick1G we have a branch which tested this, last i checked there seemed to be an issue with the availability, we will recheck the status and let you know
